libs/langgraph/langgraph/pregel/runner.py

```diff
@@ -412,9 +406,27 @@ def call(
             assert fut is not None, "writer did not return a future for call"
             # return a chained future to ensure commit() callback is called
             # before the returned future is resolved, to ensure stream order etc
-            sfut: asyncio.Future[Any] = asyncio.Future(loop=loop)
-            chain_future(fut, sfut)
-            return sfut
+            try:
+                asyncio.current_task()
+                in_async = True
+            except RuntimeError:
+                in_async = False
+            # if in async context return an async future
+            # otherwise return a chained sync future
+            if in_async:
+                if isinstance(fut, asyncio.Task):
+                    sfut: Union[asyncio.Future[Any], concurrent.futures.Future[Any]] = (
+                        asyncio.Future(loop=loop)
```

Comment:
Took me a moment to find where loop is defined in the outer context after the fn definition. Would have been easier for me to read if it were passed in as an arg fwiw